evm/bridge-token-factory/contracts/BridgeTokenFactory.sol

```diff
@@ -161,28 +171,43 @@ contract BridgeTokenFactory is
     function finTransferExtension(BridgeTypes.FinTransferPayload memory payload) internal virtual {}
 
     function initTransfer(
-        string calldata token,
+        address tokenAddress,
         uint128 amount,
         uint128 fee,
         uint128 nativeFee,
-        string calldata recipient
+        string calldata recipient,
+        string calldata message
     ) payable external whenNotPaused(PAUSED_INIT_TRANSFER) {
         initTransferNonce += 1;
-        require(isBridgeToken[nearToEthToken[token]], "ERR_NOT_BRIDGE_TOKEN");
         if (fee >= amount) {
             revert InvalidFee();
         }
 
-        address tokenAddress = nearToEthToken[token];
+        uint256 extensionValue = msg.value - nativeFee;
```

Comment:
Still feels like calculating value inside the extension method is more appropriate

Reply:
Nope, the difference should be calculated here for two reasons: 
1- we could implement a refund later.
2- it works as a verification of `msg.value >= nativeFee `